cpp/src/replace/replace.cu

```diff
@@ -13,97 +31,840 @@
+      }
+    } else {
+      if (replacement_values.has_nulls()) {
+        replace = replace_kernel<col_type, false, true>;
+      } else {
+        replace = replace_kernel<col_type, false, false>;
+      }
+    }
+
+    std::unique_ptr<cudf::column> output;
+    if (input_col.has_nulls() || replacement_values.has_nulls()) {
+      output = cudf::experimental::allocate_like(input_col,
+                                                 cudf::experimental::mask_allocation_policy::ALWAYS,
+                                                 mr);
+    }
+    else {
+      output = cudf::experimental::allocate_like(input_col,
+                                                 cudf::experimental::mask_allocation_policy::NEVER,
+                                                 mr);
+    }
+
+    cudf::mutable_column_view outputView = output->mutable_view();
+
+    cudf::experimental::detail::grid_1d grid { outputView.size(), BLOCK_SIZE, 1 };
+
+    auto device_in = cudf::column_device_view::create(input_col);
+    auto device_out = cudf::mutable_column_device_view::create(outputView);
+    auto device_values_to_replace = cudf::column_device_view::create(values_to_replace);
+    auto device_replacement_values = cudf::column_device_view::create(replacement_values);
+
+    replace<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(
+                                                        *device_in,
+                                                        *device_out,
+                                                        valid_count,
+                                                        outputView.size(),
+                                                        *device_values_to_replace,
+                                                        *device_replacement_values);
+
+    cudaStreamSynchronize(stream);
+
+    if (outputView.nullable()) {
+      output->set_null_count(output->size() - valid_counter.value());
+    }
+    return output;
+  }
+  template<typename col_type,
+      std::enable_if_t<not cudf::is_fixed_width<col_type>()>* = nullptr>
+  std::unique_ptr<cudf::column> operator()(cudf::column_view const& input_col,
+                                           cudf::column_view const& values_to_replace,
+                                           cudf::column_view const& replacement_values,
+                                           rmm::mr::device_memory_resource* mr,
+                                           cudaStream_t stream = 0) {
+    rmm::device_scalar<cudf::size_type> valid_counter(0);
+    cudf::size_type *valid_count = valid_counter.data();
+
+    auto replace_first = replace_strings_first_pass<true, true>;
+    auto replace_second = replace_strings_second_pass<true,true>;
+    if (input_col.has_nulls()) {
+      if (replacement_values.has_nulls()) {
+        replace_first = replace_strings_first_pass<true, true>;
+        replace_second = replace_strings_second_pass<true,true>;
+      } else {
```

Comment:
We can remove this else and make it the default values line 414 and 415, one less else.

Reply:
Done.